### PR TITLE
feat: Enable destinations to completely skip migration that are not supported

### PR DESCRIPTION
**Pull request by bbernays**
#### Summary

Some destinations don't support all migrations types. This allows them to opt out

### CODE REVIEW

File: plugin/testing_write_migrate.go [hunk 1]
```diff
@@ -149,6 +152,9 @@ func (s *WriterTestSuite) testMigrate(
 	})
 
 	t.Run("add_column_not_null"+suffix, func(t *testing.T) {
+		if s.tests.SkipSpecificMigrations.AddColumn {
```
Comment (erezrokah) [inline plugin/testing_write_migrate.go:155]:
```suggestion
		if s.tests.SkipSpecificMigrations.AddColumnNotNull {
```

File: plugin/testing_write_migrate.go [hunk 2]
```diff
@@ -200,6 +209,9 @@ func (s *WriterTestSuite) testMigrate(
 	})
 
 	t.Run("remove_column_not_null"+suffix, func(t *testing.T) {
+		if s.tests.SkipSpecificMigrations.RemoveColumn {
```
Comment (erezrokah) [inline plugin/testing_write_migrate.go:212]:
```suggestion
		if s.tests.SkipSpecificMigrations.RemoveColumnNotNull {
```

File: plugin/testing_write.go [hunk 3]
```diff
@@ -48,6 +48,17 @@ type SafeMigrations struct {
 	MovePKToCQOnly         bool
 }
 
+// Migrations defines which migrations should be skipped completely
+type Migrations struct {
+	AddColumn              bool
+	AddColumnNotNull       bool
+	RemoveColumn           bool
+	RemoveColumnNotNull    bool
+	ChangeColumn           bool
```
Comment (erezrokah) [inline plugin/testing_write.go:57]:
```suggestion
```

Aren't all migrations support these? Maybe only allow the new ones?

Comment (bbernays) [reply plugin/testing_write.go:57]: My reasoning was that new destinations might not support those migrations and so might as well expose them for future usage.

Comment (erezrokah) [reply plugin/testing_write.go:57]: I think the baseline should be to support these basic changes in all destinations. I'd only open this for configuration if we run into a destination that really needs them

Comment (bbernays) [reply plugin/testing_write.go]: fixed in https://github.com/cloudquery/plugin-sdk/pull/1560/commits/b774c094ad9d394062bdffca8afec5dc7d623cf0

File: plugin/testing_write_migrate.go [hunk 4]
```diff
@@ -270,13 +270,16 @@ func (s *WriterTestSuite) testMigrate(
 				{Name: "uuid", Type: types.ExtensionTypes.UUID},
 				{Name: "bool", Type: arrow.FixedWidthTypes.Boolean, NotNull: true},
 			}}
-		require.NoError(t, s.migrate(ctx, target, source, s.tests.SafeMigrations.RemoveUniqueConstraint, forceMigrate))
+		require.NoError(t, s.migrate(ctx, target, source, s.tests.SafeMigrations.MovePKToCQOnly, forceMigrate))
```
Comment (erezrokah) [inline plugin/testing_write_migrate.go:273]: Was this change intentional?

**Comment by erezrokah** [comment]
For the destinations that don't support these migrations (safe or forced) how would that look like from a user perspective?

**Comment by bbernays** [comment]
> For the destinations that don't support these migrations (safe or forced) how would that look like from a user perspective?

For a forced migration, if a destination cannot detect that there is a difference between the before and after then the destination will just insert the data rather than dropping first

**Review by erezrokah — APPROVED** [review]
